test/functional/collection_tests.js

```diff
@@ -12,557 +11,419 @@ describe('Collection', function() {
+                      expect(result).to.be.true;
+                      client.close(done);
+                    });
+                  }
+                );
               });
             });
           });
         });
-      }
+      });
     });
+
     /**
      * @ignore
      */
-    it('should correctly list back collection names containing .', {
-      metadata: {
-        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-      },
-
-      // The actual test we wish to run
-      test: function(done) {
-        client.connect((err, client) => {
+    it('should correctly list back collection names containing .', function(done) {
+      db.createCollection('test.game', (err, collection) => {
+        // Verify that all the result are correct coming back (should contain the value ok)
+        expect('test.game').to.equal(collection.collectionName);
+        // Let's check that the collection was created correctly
+        db.listCollections().toArray((err, documents) => {
           expect(err).to.not.exist;
-          const db1 = client.db('test');
-          db1.createCollection('test.game', (err, collection) => {
-            // Verify that all the result are correct coming back (should contain the value ok)
-            expect('test.game').to.equal(collection.collectionName);
-            // Let's check that the collection was created correctly
-            db1.listCollections().toArray((err, documents) => {
-              expect(err).to.not.exist;
-              let found = false;
-              documents.forEach(x => {
-                if (x.name === 'test.game') found = true;
-              });
-
-              expect(found).to.be.true;
-              client.close();
-              done();
-            });
+          let found = false;
+          documents.forEach(x => {
+            if (x.name === 'test.game') found = true;
           });
+
+          expect(found).to.be.true;
+          client.close(done);
         });
-      }
+      });
     });
 
     /**
      * @ignore
      */
-    it('should access to collections', {
-      metadata: {
-        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-      },
-
-      // The actual test we wish to run
-      test: function(done) {
-        client.connect((err, client) => {
-          expect(err).to.not.exist;
-          const db = client.db(this.configuration.db);
-          // Create two collections
-          db.createCollection('test.spiderman', () => {
-            db.createCollection('test.mario', () => {
-              // Insert test documents (creates collections)
-              db.collection('test.spiderman', (err, spiderman_collection) => {
-                spiderman_collection.insert(
-                  { foo: 5 },
-                  this.configuration.writeConcernMax(),
-                  err => {
+    it('should access to collections', function(done) {
```

Comment:
If we can come up with a better test name here, that would be great!

Reply:
renamed to `should correctly access collection names`